flask_rebar/request_utils.py

```diff
@@ -52,22 +52,20 @@ def __getitem__(self, key):
         return self.headers[key]
 
 
-def get_json_from_resp(resp):
-    return json.loads(resp.data.decode("utf-8"))
-
-
-def response(data, status_code=200, headers=None):
+def response(data, status_code=200, headers=None, mimetype=None):
     """
     Constructs a flask.jsonify response.
 
     :param dict data: The JSON body of the response
     :param int status_code: HTTP status code to use in the response
+    :param dict headers: Additional headers to attach to the response
+    :param str mimetype: Default Content-Type response header
     :rtype: flask.Response
     """
-    resp = jsonify(data)
-
-    if not get_json_from_resp(resp=resp):
-        resp.data = ""
+    if data is None:
+        resp = Response(mimetype=mimetype)
+    else:
+        resp = jsonify(data)
```

Comment:
I think we should use `mimetype` in all cases, not just if `data is None`. As a developer, if I set the `mimetype` on a handler, I would expect it to return that mimetype.